### PR TITLE
GOVUKAPP-594 Update Recommendation

### PR DESCRIPTION
**Pull request by TimGJD**
# Title
Update Recommendation

# Description
Add Update Recommendation screen and logic to show it

## JIRA ticket(s)
  - [GOVAPP-594](https://govukverify.atlassian.net/browse/GOVUKAPP-594)

## Figma
  - [Figma board](https://www.figma.com/design/5u5cvVCgKY3Rng2ADNV30K/2024-07---GOV.UK-app---beta-designs?node-id=363-6730&node-type=section&t=Y6vBXaDTHQQfnkOD-0)

| Light Mode | Dark Mode |
|---|---|
| Screenshot_1731421523 | Screenshot_1731421501 |

### CODE REVIEW

File: app/src/main/kotlin/uk/govuk/app/navigation/RecommendUpdateNavigation.kt
Comment (davidc-gds) [inline app/src/main/kotlin/uk/govuk/app/navigation/RecommendUpdateNavigation.kt:1]: I'm not sure we need dedicated graphs for this (or the unavailable screen). I feel like we can just add them as a composable. What do you think @TimGJD @gclssvglx ?

File: config/src/main/kotlin/uk/govuk/app/config/extensions/StringExtension.kt
Comment (davidc-gds) [inline config/src/main/kotlin/uk/govuk/app/config/extensions/StringExtension.kt:1]: Will (or perhaps should) this work  if e.g. we compare 1.0 with 1.0.1?

Comment (TimGJD) [reply config/src/main/kotlin/uk/govuk/app/config/extensions/StringExtension.kt:1]: I ran a test with your params and it passed true that 1.0 is less than 1.0.1:

image

Have you spotted any flaws in the function?

Comment (davidc-gds) [reply config/src/main/kotlin/uk/govuk/app/config/extensions/StringExtension.kt:1]: Nope, just couldn't figure out at a glance if that would be the case.

File: app/src/main/kotlin/uk/govuk/app/navigation/RecommendUpdateNavigation.kt
Comment (TimGJD) [reply app/src/main/kotlin/uk/govuk/app/navigation/RecommendUpdateNavigation.kt:1]: Ok, I will take a look at making this change

Comment (TimGJD) [reply app/src/main/kotlin/uk/govuk/app/navigation/RecommendUpdateNavigation.kt:1]: Agree @davidc-gds the navigation graphs were unnecessary.
Done

**Review by davidc-gds — APPROVED** [review]

**Review by davidc-gds — APPROVED** [review]

**Pull request merged**